test/concourse/model/traits/filter_test.py

```diff
@@ -0,0 +1,140 @@
+import dataclasses
+import typing
+
+import pytest
+
+import concourse.model.traits.filter as examinee
+
+
+# Dummy dataclass for testing
+@dataclasses.dataclass
+class Dummy:
+    name: str
+
+
+@pytest.fixture
+def dummy():
+
+    def _dummy(
+        name: str = 'Dummy',
+    ):
+        return Dummy(
+            name=name,
+        )
+
+    return _dummy
+
+
+class TestMatchingFilter:
+    @pytest.fixture
+    def rule(self):
+        def _rule(
+            target: str = '',
+            expression: str = '',
+            matching_semantics: str = 'include',
+        ):
+            return examinee.MatchingConfigRule(
+                target=target,
+                expression=expression,
+                matching_semantics=examinee.ComponentFilterSemantics(
+                    matching_semantics
+                ),
+            )
+
+        return _rule
+
+    @pytest.fixture
+    def matching_config(self):
+        def _matching_config(
+            name: str = 'TestConfig',
+            rules: typing.List[examinee.MatchingConfigRule] = [],
+        ):
+            return examinee.MatchingConfig(
+                name=name,
+                rules=rules,
+            )
+
+        return _matching_config
+
+    def test_unspecific_target_fails(self, dummy, matching_config, rule):
+        config = matching_config(
```

Comment:
hmmm.... is this actually that much shorter than:
```
MatchingConfig(
 name='some-name',
 rules=(
  MatchingConfigRule(
   target='tgt',
  expression='some-expression',
  matching_semantics='include',
 ),
)
```

by line-count, I think this is not the case; it would make the test a tiny bit easier to understand, because there would be less abstraction from the actual code, though.